test/new-e2e/tests/containers/docker/docker_test.go

```diff
@@ -0,0 +1,141 @@
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/infra"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/containers"
+	dockervm "github.com/DataDog/test-infra-definitions/scenarios/aws/dockerVM"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/stretchr/testify/suite"
+)
+
+type DockerSuite struct {
+	containers.BaseSuite
+	isDsdStandalone bool
+	stackName       string
+}
+
+func TestDockerSuite(t *testing.T) {
+	// Full Agent
+	suite.Run(t, &DockerSuite{stackName: "docker-stack", isDsdStandalone: false})
+
+	// DogstatsD Standalone
+	suite.Run(t, &DockerSuite{stackName: "docker-stack", isDsdStandalone: true})
+}
+
+func (suite *DockerSuite) getAgentImage() string {
+	if suite.isDsdStandalone {
+		return "datadog/dogstatsd:latest"
+	}
+
+	return ""
+}
+
+func (suite *DockerSuite) SetupSuite() {
+
+	ctx := context.Background()
+
+	stackConfig := runner.ConfigMap{
+		"ddagent:deploy":        auto.ConfigValue{Value: "true"},
+		"ddagent:fakeintake":    auto.ConfigValue{Value: "true"},
+		"ddagent:fullImagePath": auto.ConfigValue{Value: suite.getAgentImage()},
+	}
+
+	if suite.isDsdStandalone {
+		stackConfig["ddagent:fullImagePath"] = auto.ConfigValue{Value: "datadog/dogstatsd:latest"}
+	}
+
+	_, stackOutput, err := infra.GetStackManager().GetStackNoDeleteOnFailure(ctx, suite.stackName, stackConfig, dockervm.Run, false)
+
+	if !suite.Assert().NoError(err) {
+		suite.FailNow(ctx)
+	}
+
+	vmConnection := stackOutput.Outputs["vm-connection"].Value.(map[string]interface{})
+	suite.stackName = fmt.Sprintf("%s-%v", os.Getenv("USER"), vmConnection["host"])
+
+	fakeintakeHost := stackOutput.Outputs["fakeintake-host"].Value
+	suite.Fakeintake = fakeintake.NewClient(fmt.Sprintf("http://%s", fakeintakeHost))
+
+	suite.Eventuallyf(
+		func() bool { return suite.Fakeintake.GetServerHealth() == nil },
+		1*time.Minute,
+		5*time.Second,
+		"Fakeintake is never healthy",
+	)
+
+	err = suite.Fakeintake.FlushServerAndResetAggregators()
+	if !suite.Assert().NoError(err) {
+		fmt.Println("Failed to flush/reset fakeintake")
+		suite.FailNow(ctx)
+	}
+
+	suite.BaseSuite.SetupSuite()
+}
+
+func (suite *DockerSuite) FailNow(ctx context.Context) {
+	_, err := infra.GetStackManager().GetPulumiStackName(suite.stackName)
+	suite.Require().NoError(err)
+	if !runner.GetProfile().AllowDevMode() || !*containers.KeepStacks {
+		infra.GetStackManager().DeleteStack(ctx, suite.stackName)
+	}
+	suite.T().FailNow()
+}
+
+func (suite *DockerSuite) TearDownSuite() {
+	suite.BaseSuite.TearDownSuite()
+}
+
+func (suite *DockerSuite) TestDSDWithUDS() {
+	suite.AssertMetric(&containers.TestMetricArgs{
+		Filter: containers.TestMetricFilterArgs{
+			Name: "custom.metric.uds",
+		},
+		Expect: containers.TestMetricExpectArgs{
+			Tags: &[]string{
+				`^container_name:metric-sender-uds$`,
+				`^short_image`,
+				`^image_name`,
+				`^image_id`,
+				`^docker_image`,
+				`^container_id`,
+			},
```

Comment:
I see this set of tags repeated in other tests. Would it make sense to unify this assertion, maybe to make it easier to keep in sync across different scenarios, and changes in the code that produces this metric?

Reply:
Yes there is an overlap between tests when it comes to what tags we are asserting. 

But, I don't think it will add much value to unify the list of tags we are asserting. We are already unifying the assertion logic by using the `baseSuite` methods. I believe having the tags explicitly listed as such is easier to quickly know what the test is checking for, and I think it is highly unlikely that we need to change tags on all tests.